File: repos_estudiantes_ALVAREZ_primera_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/Polimorfismo/Ejercicio1/VideoJuego.py
class Videojuego:
    def __init__(self, nombre="sinNombre", plataforma="sinPlataforma", cantidad_jugadores=0):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidad_jugadores = cantidad_jugadores

    def mostrar(self):
        print(f"Nombre: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Cantidad de Jugadores: {self.cantidad_jugadores}")

    def agregar_un_jugador(self):
        self.cantidad_jugadores += 1
        print(f"Se agregó 1 jugador. Total jugadores: {self.cantidad_jugadores}")

    def agregar_varios_jugadores(self, cantidad):
        self.cantidad_jugadores += cantidad
        print(f"Se agregaron {cantidad} jugadores. Total jugadores: {self.cantidad_jugadores}")


if __name__ == "__main__":
    juego1 = Videojuego("Left 4 Dead", "Steam", 4)
    juego2 = Videojuego("Call of Duty", "PC")

    # Agregar jugadores
    juego1.agregar_un_jugador()
    juego1.agregar_varios_jugadores(3)
    juego2.agregar_un_jugador()
    juego2.agregar_varios_jugadores(5)

    # Mostrar información de los videojuegos
    juego1.mostrar()
    juego2.mostrar()

File: repos_estudiantes_ALVAREZ_primera_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/Polimorfismo/Ejercicio1/Videojuego.java

package Ejercicio1;

public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;
    //b) Sobrecargar el constructor 2 veces
    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }
    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 0;
    }
    public Videojuego() {
        this.nombre = "sinNombre";
        this.plataforma = "sinPlataforma";
        this.cantidadJugadores = 0;
    }
    //c) Implementar un método mostrar()
    public void mostrar() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Plataforma: " + plataforma);
        System.out.println("Cantidad de Jugadores: " + cantidadJugadores);
    }
    
    //d) Sobrecargar el método agregarJugadores() donde en el primero se agregue
    //solo 1 jugador y en otro se ingrese una cantidad de jugadores a aumentar.
    public void agregarJugadores() {
        this.cantidadJugadores += 1;
        System.out.println("Se agregó 1 jugador. Total jugadores: " + cantidadJugadores);
    }
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
        System.out.println("Se agregaron " + cantidad + " jugadores. Total jugadores: " + cantidadJugadores);
    }
    public static void main(String[] args) {
        //a) Instanciar al menos 2 videojuegos
        Videojuego juego1 = new Videojuego("Left 4 Dead", "Steam", 4);
        Videojuego juego2 = new Videojuego("Call of Duty", "PC");
        juego1.agregarJugadores();
        juego1.agregarJugadores(3);
        juego2.agregarJugadores();
        juego2.agregarJugadores(5);
        juego1.mostrar();
        juego2.mostrar();
    }
}